cli/verify.go

```diff
@@ -0,0 +1,106 @@
+package cli
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/docker/cli/cli/config/configfile"
+	"github.com/docker/cli/cli/config/credentials"
+	"github.com/docker/cli/cli/config/types"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/cmd/cosign/cli/verify"
+)
+
+// Verify performs signature verification of keyless signatures
+func verifyCollection(o *PullOptions, ctx context.Context) error {
+
+	regopts := options.RegistryOptions{
+		Keychain: authn.DefaultKeychain,
+	}
+	if o.PlainHTTP || o.Insecure {
+		regopts.AllowInsecure = true
+	}
+
+	if len(o.Configs) != 0 {
```

Comment:
One more change, and I think this is good to go. Let's move this logic into a new function so it can be reused.